include/envoy/tracing/http_tracer.h

```diff
@@ -158,6 +158,20 @@ class Span {
    * @param sampled whether the span and any subsequent child spans should be sampled
    */
   virtual void setSampled(bool sampled) PURE;
+
+  /**
+   * Retrieve a key's value from the current span's baggage
+   * @param key baggage key
+   * @return the baggage's value for the given input key
+   */
+  virtual std::string getBaggage(const std::string& key) PURE;
+
+  /**
+   * Set a key/value pair in the current span's baggage
+   * @param key baggage key
+   * @param key baggage value
+   */
+  virtual void setBaggage(const std::string& key, const std::string& value) PURE;
```

Comment:
What's the difference with setTag?

Reply:
baggage is persisted in the `SpanContext`, which means that any subsequent spans can access data in the baggage (it gets injected/extracted across process boundaries). This is useful when you need to embed information that is available for the entirety of the request

more reading: https://opentracing.io/docs/overview/tags-logs-baggage/, https://github.com/opentracing/specification/blob/master/specification.md#set-a-baggage-item